<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="description" content="nudup : ">

    <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">

    <title>nudup</title>
  </head>

  <body>

    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/nugentechnologies/nudup">View on GitHub</a>

          <h1 id="project_title">nudup</h1>
          <h2 id="project_tagline"></h2>

            <section id="downloads">
              <a class="zip_download_link" href="https://github.com/nugentechnologies/nudup/zipball/master">Download this project as a .zip file</a>
              <a class="tar_download_link" href="https://github.com/nugentechnologies/nudup/tarball/master">Download this project as a tar.gz file</a>
            </section>
        </header>
    </div>

    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
        <h1>
<a id="nudup----nugens-tool-for-deduplicating-reads-using-n6-sequence-information" class="anchor" href="#nudup----nugens-tool-for-deduplicating-reads-using-n6-sequence-information" aria-hidden="true"><span class="octicon octicon-link"></span></a>nudup -- NuGEN's tool for deduplicating reads using N6 sequence information</h1>

<h2>
<a id="script-requires" class="anchor" href="#script-requires" aria-hidden="true"><span class="octicon octicon-link"></span></a>Script requires</h2>

<ul>
<li>samtools-0.1.18 or higher <a href="http://samtools.sourceforge.net/">samtools page</a> Tested on 0.1.18</li>
<li>python2.7 <a href="http://continuum.io/downloads">anaconda page</a> Tested on 2.7.7 </li>
<li>GNU-coreutils gzip,sed,cut,grep etc. Standard on Ubuntu 12.04+.</li>
</ul>

<h2>
<a id="usage" class="anchor" href="#usage" aria-hidden="true"><span class="octicon octicon-link"></span></a>Usage</h2>

<p>Run <code>python nudup.py -h</code> for latest usage.</p>

<pre><code>usage: nudup.py [-2] [-f INDEX.fq|READ.fq] [-o OUT_PREFIX] [-s START]
                [-l LENGTH] [-v] [-h]
                IN.sam|IN.bam

Marks/removes PCR duplicates using the N6 sequence technology used in
NuGEN's Ovation Target Enrichment libraries.

For single end reads, duplicates are marked if they fulfill the following
criteria: a) start in the same location b) have the same orientation/strand
c) have the same N6 sequence. The read with the highest mapping quality is kept
as the non-duplicate read. For paired end reads, duplicates are marked if they
fulfill the following criteria: a) start in the same location b) have the same
template length c) have the same N6 sequence. The read pair with the highest
mapping quality is kept as the non-duplicate read.

The runtime for marking and removal of duplicates depends on the format of the
inputs. Here are the two cases for running this tool:

- Case 1 (faster runtime): User supplies pre-built SAM/BAM alignment file that
  is sorted and has fixed length N6 sequence appended to the read name
- Case 2 (slower runtime): User supplies SAM/BAM alignment file and FASTQ file
  containing N6 sequence. SAM/BAM records do not have a N6 sequence, but the N6
  sequence is a substring in the read or title of the FASTQ file. FASTQ read
  names must be unique, and each SAM/BAM record must have corresponding read
  name in the FASTQ file.

Author: Anand Patel
Contact: NuGEN Technologies Inc., techserv@nugen.com

Input:
  IN.sam|IN.bam         input sorted/unsorted SAM/BAM

Options:
  -2, --paired-end      use paired end deduping with template. SAM/BAM
                        alignment must contain paired end reads.
  -f INDEX.fq|READ.fq   FASTQ file paired with input SAM/BAM (REQUIRED if
                        SAM/BAM does not have N6 sequence appended to
                        qname/read title)
  -o OUT_PREFIX, --out OUT_PREFIX
                        prefix of output file paths for sorted BAMs (default
                        will create prefix.sorted.markdup.bam,
                        prefix.sorted.dedup.bam, prefix_dup_log.txt)
  -s START, --start START
                        position in index read where N6 sequence begins. This
                        should be a 1-based value that counts in from the END
                        of the read. (default = 6)
  -l LENGTH, --length LENGTH
                        length of N6 sequence (default = 6)
  -v, --version         show program's version number and exit
  -h, --help            show this help message and exit
</code></pre>
      </section>
    </div>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <p class="copyright">nudup maintained by <a href="https://github.com/nugentechnologies">nugentechnologies</a></p>
        <p>Published with <a href="https://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

    

  </body>
</html>
