<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <link href='https://fonts.googleapis.com/css?family=Chivo:900' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-dark.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <title>nudup by nugentechnologies</title>
  </head>

  <body>
    <div id="container">
      <div class="inner">

        <header>
          <h1>nudup</h1>
          <h2></h2>
        </header>

        <section id="downloads" class="clearfix">
          <a href="https://github.com/nugentechnologies/nudup/zipball/master" id="download-zip" class="button"><span>Download .zip</span></a>
          <a href="https://github.com/nugentechnologies/nudup/tarball/master" id="download-tar-gz" class="button"><span>Download .tar.gz</span></a>
          <a href="https://github.com/nugentechnologies/nudup" id="view-on-github" class="button"><span>View on GitHub</span></a>
        </section>

        <hr>

        <section id="main_content">
          <h1>
<a id="nudup----nugens-tool-for-deduplicating-reads-using-n6-sequence-information" class="anchor" href="#nudup----nugens-tool-for-deduplicating-reads-using-n6-sequence-information" aria-hidden="true"><span class="octicon octicon-link"></span></a>nudup -- NuGEN's tool for deduplicating reads using N6 sequence information</h1>

<h2>
<a id="script-requires" class="anchor" href="#script-requires" aria-hidden="true"><span class="octicon octicon-link"></span></a>Script requires</h2>

<ul>
<li>samtools-0.1.18 or higher <a href="http://samtools.sourceforge.net/">samtools page</a> Tested on 0.1.18</li>
<li>python2.7 <a href="http://continuum.io/downloads">anaconda page</a> Tested on 2.7.7 </li>
<li>GNU-coreutils gzip,sed,cut,grep etc. Standard on Ubuntu 12.04+.</li>
</ul>

<h2>
<a id="usage" class="anchor" href="#usage" aria-hidden="true"><span class="octicon octicon-link"></span></a>Usage</h2>

<p>Run <code>python nudup.py -h</code> for latest usage.</p>

<pre><code>usage: nudup.py [-2] [-f INDEX.fq|READ.fq] [-o OUT_PREFIX] [-s START]
                [-l LENGTH] [-v] [-h]
                IN.sam|IN.bam

Marks/removes PCR duplicates using the N6 sequence technology used in
NuGEN's Ovation Target Enrichment libraries.

For single end reads, duplicates are marked if they fulfill the following
criteria: a) start in the same location b) have the same orientation/strand
c) have the same N6 sequence. The read with the highest mapping quality is kept
as the non-duplicate read. For paired end reads, duplicates are marked if they
fulfill the following criteria: a) start in the same location b) have the same
template length c) have the same N6 sequence. The read pair with the highest
mapping quality is kept as the non-duplicate read.

The runtime for marking and removal of duplicates depends on the format of the
inputs. Here are the two cases for running this tool:

- Case 1 (faster runtime): User supplies pre-built SAM/BAM alignment file that
  is sorted and has fixed length N6 sequence appended to the read name
- Case 2 (slower runtime): User supplies SAM/BAM alignment file and FASTQ file
  containing N6 sequence. SAM/BAM records do not have a N6 sequence, but the N6
  sequence is a substring in the read or title of the FASTQ file. FASTQ read
  names must be unique, and each SAM/BAM record must have corresponding read
  name in the FASTQ file.

Author: Anand Patel
Contact: NuGEN Technologies Inc., techserv@nugen.com

Input:
  IN.sam|IN.bam         input sorted/unsorted SAM/BAM

Options:
  -2, --paired-end      use paired end deduping with template. SAM/BAM
                        alignment must contain paired end reads.
  -f INDEX.fq|READ.fq   FASTQ file paired with input SAM/BAM (REQUIRED if
                        SAM/BAM does not have N6 sequence appended to
                        qname/read title)
  -o OUT_PREFIX, --out OUT_PREFIX
                        prefix of output file paths for sorted BAMs (default
                        will create prefix.sorted.markdup.bam,
                        prefix.sorted.dedup.bam, prefix_dup_log.txt)
  -s START, --start START
                        position in index read where N6 sequence begins. This
                        should be a 1-based value that counts in from the END
                        of the read. (default = 6)
  -l LENGTH, --length LENGTH
                        length of N6 sequence (default = 6)
  -v, --version         show program's version number and exit
  -h, --help            show this help message and exit
</code></pre>
        </section>

        <footer>
          nudup is maintained by <a href="https://github.com/nugentechnologies">nugentechnologies</a><br>
          This page was generated by <a href="https://pages.github.com">GitHub Pages</a>. Tactile theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.
        </footer>

        
      </div>
    </div>
  </body>
</html>
